### PR TITLE
Bah 3330 || Implemented swagger for api documentation and do some changes in readme.

### PR DESCRIPTION
**Pull request by anubhavBeehyv**
- Implemented swagger for api documentation and do some changes in readme.

### CODE REVIEW

**Review by angshu — APPROVED** [review]

File: src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java [hunk 1]
```diff
@@ -52,6 +51,9 @@ public ResponseEntity send(String phoneNumber, String messageText) {
             String jsonObject = Obj.writeValueAsString(smsRequest);
             StringEntity params = new StringEntity(jsonObject);
             request.addHeader("content-type", "application/json");
+            if(smsProperties.getToken().isEmpty()){
+                logger.warn("token doesn't exist in env");
```
Comment (mohan-13) [inline src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java]: Along with this logging statement, do add logs after executing the D7 SMS API call.
Without logging the response, we will not be able to know whether the sms has flown out from D7 or not.

File: src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java [hunk 2]
```diff
@@ -52,9 +51,16 @@ public ResponseEntity send(String phoneNumber, String messageText) {
...
+                logger.warn("token doesn't exist in environment variable");
+                return new ResponseEntity<>("token doesn't exist in environment variable" , HttpStatus.INTERNAL_SERVER_ERROR);
+            }
             request.addHeader("Authorization", "Bearer " + smsProperties.getToken());
             request.setEntity(params);
             HttpResponse response = httpClient.execute(request);
+            if(response.getStatusLine().getStatusCode()==HttpStatus.NOT_FOUND.value()){
+                logger.warn("Invalid API URL");
+            }
             return new ResponseEntity<>(response.getStatusLine().getReasonPhrase(), HttpStatus.valueOf(response.getStatusLine().getStatusCode()));
```
Comment (mohan-13) [inline src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java:71]: Can we also add a statement for sucess scenario as well ?

Comment (anubhavBeehyv) [reply src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java:71]: @mohan-13 we can do that but after sending message evertime D7 api gives same response only regardless of phone number being correct or not so I was not sure printing the same response everytime makes sense or not thats why I didn't added that log statement .

File: src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java [hunk 3]
```diff
@@ -35,12 +34,17 @@ public ResponseEntity send(String phoneNumber, String messageText) {
         logger.info("Sending SMS for ********" + phoneNumber.substring(phoneNumber.length() - 2));
         try {
             HttpClient httpClient = HttpClientBuilder.create().build();
-            HttpPost request = new HttpPost(SEND);
+            HttpPost request = new HttpPost(smsProperties.getProviderApi());
             Message message = new Message();
             message.setChannel("sms");
             message.setMsg_type("text");
+
+            if ((phoneNumber.length()==10 || phoneNumber.length()==11) && !phoneNumber.startsWith("+")) {
```
Comment (mohan-13) [inline src/main/java/org/bahmni/sms/impl/DefaultSmsSender.java:42]: Why is the length based check required here ? Will this work for different country phone numbers ?

**Pull request merged**